feat: ui.progress_bar and ui.progress_circle (#892)
deephaven/deephaven-plugins: plugins/ui/docs/components/progress_bar.md

# Progress Bar

Progress Bars show the progression of a system operation: downloading, uploading, processing, etc., in a visual way. They can represent either determinate or indeterminate progress.

## Example

```python
from deephaven import ui

@ui.component
def ui_progress_bar():

>     return ui.progress_bar(size="L", is_indeterminate=True)

Reviewer: ```suggestion
    return ui.progress_bar(is_indeterminate=True)
```

progress_bar = ui_progress_bar()
```

## UI Recommendations

1. Use the appropriate size based on the parent's size.
2. Use `static_color="white"` or `static_color="black"` if necessary to ensure the progress circle has enough contrast with the background.
3. If the value of the progress is unknown, use `is_indeterminate=True`.

## Value

The progress is controlled by the `value`, `min_value`, and `max_value` props. The default values of `min_value` and `max_value` are `0` and `100`, respectively. 

```python

> def value_variants():
>     return ui.flex(
>         ui.progress_bar(value=50),
>         ui.progress_bar(value=50, min_value=25, max_value=125),
>         align_items="start",
>         direction="column",
>         row_gap="20px",
>     )
> 
> 
> progress_bar_value_examples = value_variants()

Reviewer: ```suggestion
@ui.component
def value_variants():
    return [
        ui.progress_bar(value=50),
        ui.progress_bar(value=50, min_value=25, max_value=125),
    ]

progress_bar_value_examples = value_variants()
```

Similar output with less code.

```

## Indeterminate

Use `is_indeterminate=True` if the progress can not be determined.

```python

> def indeterminate_variants():
>     return ui.flex(
>         ui.progress_bar(value=70),
>         ui.progress_bar(is_indeterminate=True),
>         align_items="start",
>         direction="column",
>         row_gap="20px",
>     )

Reviewer: ```suggestion
def indeterminate_variants():
    return ui.progress_bar(is_indeterminate=True)        
```

progress_bar_indeterminate_examples = indeterminate_variants()
```

## Size

Progress Bar comes in two different sizes determined by the `size` prop: `"S"` and `"L"`. By default, the size is `"L"`.

```python

> def size_variants():

Reviewer: use @ui.component simplification shown above.

Missing L example

Author: The progress bar with no `size` prop is L - I wanted to show that L is the default. Should I explicitly add the prop?

Reviewer: acknowledgeed.

    return ui.flex(
        ui.progress_bar(value=70, size="S"),
        ui.progress_bar(value=70),
        align_items="start",
        direction="column",
        row_gap="20px",
    )

progress_bar_size_examples = size_variants()
```

## Static Color

The `static_color` prop can be used to control the color of the progress bar between the default color, `"black"`, and `"white"`.

```python
def color_variants():
    return ui.flex(
        ui.view(ui.progress_bar(value=70, margin="10px")),

>         ui.view(ui.progress_bar(value=70, static_color="white", margin="10px")),

Reviewer: add `background_color="black`

1. Use the appropriate size based on the parent's size.
2. Use `static_color="white"` or `static_color="black"` if necessary to ensure the progress circle has enough contrast with the background.
3. If the value of the progress is unknown, use `is_indeterminate=True`.

## Value

The progress is controlled by the `value`, `min_value`, and `max_value` props. The default values of `min_value` and `max_value` are `0` and `100`, respectively. 

```python
from deephaven import ui

@ui.component
def value_variants():
    return [
        ui.progress_bar(value=50),
        ui.progress_bar(value=50, min_value=25, max_value=125),
    ]

progress_bar_value_examples = value_variants()
```

## Indeterminate

Use `is_indeterminate=True` if the progress can not be determined.

```python
from deephaven import ui

@ui.component
def indeterminate_variants():
    return ui.progress_bar(is_indeterminate=True)

progress_bar_indeterminate_examples = indeterminate_variants()
```

## Size

Progress Bar comes in two different sizes determined by the `size` prop: `"S"` and `"L"`. By default, the size is `"L"`.

```python
from deephaven import ui

@ui.component
def size_variants():

>     return ui.flex(
>         ui.progress_bar(value=70, size="S"),
>         ui.progress_bar(value=70),
>         align_items="start",
>         direction="column",
>         row_gap="20px",
>     )

Reviewer: ```suggestion
    return [
        ui.progress_bar(value=70, size="S"),
        ui.progress_bar(value=70),
    ]
```

skimpily example to minimum necessary code.

# Progress Bar

Progress Bars show the progression of a system operation: downloading, uploading, processing, etc., in a visual way. They can represent either determinate or indeterminate progress.

## Example

```python
from deephaven import ui

@ui.component
def ui_progress_bar():
    return ui.progress_bar(size="L", is_indeterminate=True)

progress_bar = ui_progress_bar()
```

## UI Recommendations

> 1. Use the appropriate size based on parent's size

Reviewer 2: ```suggestion
1. Use the appropriate size based on the parent's size.
```

> 2. Use `static_color="white"` or `static_color="white"` if necessary to make sure the progress circle has enough contrast with the background

Reviewer 2: ```suggestion
2. Use `static_color="white"` or `static_color="white"` if necessary to ensure the progress circle has enough contrast with the background.
```

Reviewer 2: aren't these the same?

> 3. If the value of the progress is unknown, use `is_indeterminate=True`

Reviewer 2: ```suggestion
3. If the value of the progress is unknown, use `is_indeterminate=True`.
```

> ## Visual Options

Reviewer: Split these examples by prop please, with headings and a minimal reproducible example for each.

> Progress Bar comes in 2 different sizes by the `size` prop: `"S"` and `"L"`. Furthermore, the `static_color` prop can be used to control the color of the progress circle.

Reviewer 2: ```suggestion
Progress Bar comes in two different sizes determined by the `size` prop: `"S"` and `"L"`. Furthermore, the `static_color` prop can be used to control the color of the progress circle.
```
